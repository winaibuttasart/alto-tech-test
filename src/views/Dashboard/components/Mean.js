import React from 'react'
import { Card, CardBody } from 'reactstrap'

export default function Mean({ title, data, subfix }) {
    const mean = data.reduce((a, b) => a + b, 0) / data.length

    return (
        <Card className="card-stats">
            <CardBody>
                <p className="card-category">
                    {title}
                    <span className="right">{`${mean.toFixed(2)} ${subfix}`}</span>
                </p>
            </CardBody>
        </Card>
    )
}
