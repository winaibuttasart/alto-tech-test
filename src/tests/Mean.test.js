import React from 'react'
import { shallow, render } from 'enzyme'
import Mean from '../views/Dashboard/components/Mean'

describe('render <Mean />', () => {
    it('should render without throwing an error', () => {
        shallow(
            <Mean
                title="ค่า mean จาก Line Chart"
                data={[5, 5, 5, 5]}
                subfix="บาท"
            />
        )
    })

    it('should render Mean to static HTML', () => {
        const componentLineChart = render(
            <Mean
                title="ค่า mean จาก Line Chart "
                data={[5, 5, 5, 5]}
                subfix="บาท"
            />
        )
        const componentBarChart = render(
            <Mean
                title="ค่า mean จาก Bar Chart "
                data={[1, 2, 2, 6]}
                subfix="บาท"
            />
        )
        expect(componentLineChart.text()).toEqual('ค่า mean จาก Line Chart 5.00 บาท')
        expect(componentBarChart.text()).toEqual('ค่า mean จาก Bar Chart 2.75 บาท')
    })
})
