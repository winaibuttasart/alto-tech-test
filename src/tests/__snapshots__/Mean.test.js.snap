// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot Mean 1`] = `
<div
  className="card-stats card"
>
  <div
    className="card-body"
  >
    <p
      className="card-category"
    >
      ค่า mean จาก Line Chart
      <span
        className="right"
      >
        5.00 บาท
      </span>
    </p>
  </div>
</div>
`;
